### PR TITLE
Deprecate MethodMatcher constructors and getter methods

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Related to #1876

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
The goal is to eventually lift `MethodMatcher` to being an interface type that can be composed of multiple matchers

File: rewrite-gradle/src/main/java/org/openrewrite/gradle/AddDelegatesToGradleApi.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, Ex
                         return param;
                     }
                     // Construct a matcher that will identify methods identical to this one except they accept Action instead of Closure
-                    MethodMatcher matcher = new MethodMatcher(method.getMethodType().withParameterTypes(
+                    MethodMatcher matcher = MethodMatcher.create(method.getMethodType().withParameterTypes(
```
Comment (traceyyoshima) [inline rewrite-gradle/src/main/java/org/openrewrite/gradle/AddDelegatesToGradleApi.java:82]: A minor suggestion here, I'd rename `create` to `build` to reflect our other APIs.

Comment (JLLeitschuh) [reply rewrite-gradle/src/main/java/org/openrewrite/gradle/AddDelegatesToGradleApi.java:82]: I'm easy, happy to make that change. In my experience `build` implies a 'completion' or 'finalization' of some kind of builder pattern style API. This isn't really the builder pattern here, so i don't know if I'd use the `build` method name.

I'm happy to make the change if you still think that `build` is better. But it's not the name I'd expect as an API user.

**Comment by JLLeitschuh** [comment]
Blocked by #1883

File: rewrite-java/src/main/java/org/openrewrite/java/MethodMatcher.java [hunk 2]
```diff
@@ -62,7 +62,7 @@
  * </PRE>
  */
 @SuppressWarnings("NotNullFieldNotInitialized")
-@Getter
+@Getter(onMethod_={@Deprecated})
```
Comment (JLLeitschuh) [inline rewrite-java/src/main/java/org/openrewrite/java/MethodMatcher.java:65]: This is what breaks javadoc generation (see: #1883)

**Comment by traceyyoshima** [comment]
Hi @JLLeitschuh, I spoke with the team about the changes and we won't be accepting the PR at this time. We'll keep an ear open to listen to community feedback as a whole and may make the changes at a later time. Thanks again for the time and effort you put into the PR!

**Pull request closed without merging**